diff old/library/debug.c new/library/debug.c
--- old/library/debug.c
+++ new/library/debug.c
@@ -88,7 +88,7 @@
 
     if( debug_log_mode == POLARSSL_DEBUG_LOG_RAW )
     {
-        ssl->f_dbg( ssl->p_dbg, level, str );
+        ssl->f_dbg( ssl->p_dbg, level, text );
         return;
     }

diff old/include/polarssl/bignum.h new/include/polarssl/bignum.h
--- old/include/polarssl/bignum.h
+++ new/include/polarssl/bignum.h
@@ -69,7 +69,7 @@
  */
 #define POLARSSL_MPI_MAX_LIMBS                             10000
 
-#if !defined(POLARSSL_CONFIG_OPTIONS)
+#if !defined(POLARSSL_MPI_WINDOW_SIZE)
 /*
  * Maximum window size used for modular exponentiation. Default: 6
  * Minimum value: 1. Maximum value: 6.
@@ -80,7 +80,9 @@
  * Reduction in size, reduces speed.
  */
 #define POLARSSL_MPI_WINDOW_SIZE                           6        /**< Maximum windows size used. */
+#endif /* !POLARSSL_MPI_WINDOW_SIZE */
 
+#if !defined(POLARSSL_MPI_MAX_SIZE)
 /*
  * Maximum size of MPIs allowed in bits and bytes for user-MPIs.
  * ( Default: 512 bytes => 4096 bits, Maximum tested: 2048 bytes => 16384 bits )
@@ -89,8 +91,7 @@
  * of limbs required (POLARSSL_MPI_MAX_LIMBS) is higher.
  */
 #define POLARSSL_MPI_MAX_SIZE                              512      /**< Maximum number of bytes for usable MPIs. */
-
-#endif /* !POLARSSL_CONFIG_OPTIONS */
+#endif /* !POLARSSL_MPI_MAX_SIZE */
 
 #define POLARSSL_MPI_MAX_BITS                              ( 8 * POLARSSL_MPI_MAX_SIZE )    /**< Maximum number of bits for usable MPIs. */
 
diff old/include/polarssl/check_config.h new/include/polarssl/check_config.h
--- old/include/polarssl/check_config.h
+++ new/include/polarssl/check_config.h
@@ -84,12 +84,12 @@
 #error "POLARSSL_ENTROPY_C defined, but not all prerequisites"
 #endif
 #if defined(POLARSSL_ENTROPY_C) && defined(POLARSSL_SHA512_C) &&         \
-    defined(POLARSSL_CONFIG_OPTIONS) && (CTR_DRBG_ENTROPY_LEN > 64)
+    defined(CTR_DRBG_ENTROPY_LEN) && (CTR_DRBG_ENTROPY_LEN > 64)
 #error "CTR_DRBG_ENTROPY_LEN value too high"
 #endif
 #if defined(POLARSSL_ENTROPY_C) &&                                            \
     ( !defined(POLARSSL_SHA512_C) || defined(POLARSSL_ENTROPY_FORCE_SHA256) ) \
-    && defined(POLARSSL_CONFIG_OPTIONS) && (CTR_DRBG_ENTROPY_LEN > 32)
+    && defined(CTR_DRBG_ENTROPY_LEN) && (CTR_DRBG_ENTROPY_LEN > 32)
 #error "CTR_DRBG_ENTROPY_LEN value too high"
 #endif
 #if defined(POLARSSL_ENTROPY_C) && \
