--- mbedtls/include/mbedtls/cipher.h	2018-07-26 16:33:22.000000000 +0200
+++ mbedtls/include/mbedtls/cipher.h	2018-07-27 20:28:40.091543699 +0200
@@ -47,7 +45,8 @@
 #define MBEDTLS_CIPHER_MODE_WITH_PADDING
 #endif
 
-#if defined(MBEDTLS_ARC4_C) || defined(MBEDTLS_CIPHER_NULL_CIPHER)
+#if defined(MBEDTLS_ARC4_C) || defined(MBEDTLS_CIPHER_NULL_CIPHER) || \
+    defined(MBEDTLS_CHACHA20_C)
 #define MBEDTLS_CIPHER_MODE_STREAM
 #endif
 
